### PR TITLE
fix(lsp): Prevent lsp crash when module cannot be found

### PR DESCRIPTION
**Pull request by spotandjake**
When #1963 was added we added support for hovering over the `include` tag to see the module signature. It looks like the function used for getting the signature can sometimes throw an exception this prevents the lsp from completely crashing in that case.

This has changed the env to be the program env as `include` is always going to be on the top level and a module is always going to imported into the programs env this makes it more reliable. In the case of failure we just provide the `module Header` so its still nice.

### CODE REVIEW

File: compiler/src/language_server/hover.re [hunk 1]
```diff
@@ -126,11 +126,21 @@ let declaration_lens = (ident: Ident.t, decl: Types.type_declaration) => {
...
+      switch (Modules.get_provides(decl)) {
+      | [_, ..._] => Some(module_lens(decl))
+      | [] => None
+      };
+    }) {
+    | Not_found => None
+    };
+  switch (module_decl) {
+  | Some(mod_sig) => markdown_join(header, mod_sig)
+  | None => header
```
Comment (alex-snezhko) [inline compiler/src/language_server/hover.re:142]: Should this give a warning?

Comment (spotandjake) [reply compiler/src/language_server/hover.re:142]: I don't think its necessary.

**Review by ospencer — COMMENTED** [review]
Are we sure we want to show the module header when it crashes? That typically means that module is broken somehow, and an empty module is still a valid module signature, so this could be pretty misleading.

**Comment by spotandjake** [comment]
> Are we sure we want to show the module header when it crashes? That typically means that module is broken somehow, and an empty module is still a valid module signature, so this could be pretty misleading.

I don't like the idea of an error message though. any thoughts on how we could handle it?

**Comment by spotandjake** [comment]
This should be good now

File: compiler/src/language_server/hover.re [hunk 2]
```diff
@@ -126,11 +126,17 @@ let declaration_lens = (ident: Ident.t, decl: Types.type_declaration) => {
...
+  let header = grain_code_block("module " ++ Path.name(path));
+  let decl = Env.find_module(path, None, env);
+  let module_decl =
+    switch (Modules.get_provides(decl)) {
+    | [_, ..._] => Some(module_lens(decl))
+    | [] => None
+    };
+  switch (module_decl) {
+  | Some(mod_sig) => markdown_join(header, mod_sig)
+  | None => header
```
Comment (ospencer) [inline compiler/src/language_server/hover.re:138]: If there's no module declaration should we be showing the header?

Comment (spotandjake) [reply compiler/src/language_server/hover.re:138]: I think so. I think of a module that exists but doesn't have any exports as empty.

**Review by ospencer — APPROVED** [review]

**Pull request merged**